### PR TITLE
fix(user): add onboarding_survey enum in dashboard metadata type

### PR DESCRIPTION
**Pull request by Riddhiagrawal001**
## Type of Change

- [X] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Bug Fix: Add support for onboarding_survey enum 

### Additional Changes

- [ ] This PR modifies the API contract
- [X] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Until now, we've been utilising Calendly to gather comprehensive information regarding both merchants and users. 
After this PR we will capture all the data from the dashboard and store it in dashboard_metadata for better insights.

## How did you test it?
1. Post OnboardingSurvey enum details  
Request 
``` 
curl --location 'http://localhost:8080/user/data?keys=OnboardingSurvey' \
--header 'Authorization: JWT token' \
--header 'Content-Type: application/json' \
--data '{
    "OnboardingSurvey": {
        "designation": "some designation",
        "business_website": "some business website",
        "about_business": "about business ",
        "major_markets": [
            "Europe"
        ],
        "business_size": "business size",
        "hyperswitch_req": "hyperswitch requirement",
        "required_features": [
            "FRM",
            "Recon"
        ],
        "required_processors": [
            "Stripe"
        ],
        "planned_live_date": "planned live date",
        "miscellaneous": "some comment"
    }
}'
```

2. Get OnboardingSurvey enum details request 
```
curl --location 'http://localhost:8080/user/data?keys=OnboardingSurvey' \
--header 'Authorization: JWT token'
```
Response  
a. If data not present for the user 
`[{"OnboardingSurvey":null}]`
b. If data already present for the user 
```
[
    {
        "OnboardingSurvey": {
            "designation": "some designation",
            "about_business": "about business ",
            "business_website": "some business website",
            "hyperswitch_req": "hyperswitch requirement",
            "major_markets": [
                "Europe"
            ],
            "business_size": "business size",
            "required_features": [
                "FRM",
                "Recon"
            ],
            "required_processors": [
                "Stripe"
            ],
            "planned_live_date": "planned live date",
            "miscellaneous": "some comment"
        }
    }
]
```
## Checklist

- [X] I formatted the code `cargo +nightly fmt --all`
- [X] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by ThisIsMani — DISMISSED** [review]
Regenerate schema and check if this enum should be merchant scoped or user scoped.

File: crates/api_models/src/user/dashboard_metadata.rs [hunk 1]
```diff
@@ -46,6 +47,20 @@ pub struct ProcessorConnected {
...
+    pub designation: String,
+    pub about_business: String,
+    pub business_website: String,
+    pub hyperswitch_req: String,
+    pub major_markets: Vec<String>,
+    pub business_size: String,
+    pub required_features: Vec<String>,
+    pub required_processors: Vec<String>,
+    pub planned_live_date: String,
+    pub miscellaneous: String,
```
Comment (apoorvdixit88) [inline crates/api_models/src/user/dashboard_metadata.rs:61]: Please check once whether these parameters need to be Option

File: crates/router/src/core/user/dashboard_metadata.rs [hunk 2]
```diff
@@ -548,6 +556,17 @@ async fn insert_metadata(
             )
             .await
         }
+        types::MetaData::OnboardingSurvey(data) => {
+            utils::insert_merchant_scoped_metadata_to_db(
```
Comment (apoorvdixit88) [inline crates/router/src/core/user/dashboard_metadata.rs:560]: The enum will be one time set, not updatable?

**Review by apoorvdixit88 — CHANGES_REQUESTED** [review]

File: crates/router/src/core/user/dashboard_metadata.rs (hunk 2 above)
Comment (Riddhiagrawal001) [reply crates/router/src/core/user/dashboard_metadata.rs:560]: Adressed

**Review by apoorvdixit88 — APPROVED** [review]

**Review by ThisIsMani — APPROVED** [review]

**Comment by racnan** [comment]
- Add ***Motivation and Context***.
- Fix cURLs mentioned in description.

File: crates/api_models/src/user/dashboard_metadata.rs (hunk 1 above)
Comment (Riddhiagrawal001) [reply crates/api_models/src/user/dashboard_metadata.rs]: Adressed

**Review by racnan — APPROVED** [review]

**Pull request merged**